x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/Authentication.java

```diff
@@ -282,6 +363,16 @@ private void assertApiKeyMetadata() {
             : "API KEY authentication requires metadata to contain API KEY id, and the value must be non-null.";
     }
 
+    private void assertDomainAssignment() {
+        assert isAssignedToDomain() || domainRealms.isEmpty()
+            : "the set of realms under the same domain must be empty if realm is not assigned to any domain";
```

Comment:
If `isAssignedToDomain() == true` and `domainRealms.isEmpty() == true`, the assertion passes. But it's an invalid configuration?

Reply:
No it is not invalid. It represents the configuration where a domain only covers a single realm.